OpenTrons/opentrons-api
# fix(api): filter get_run_commands

# Overview

closes [EXEC-654](https://opentrons.atlassian.net/browse/EXEC-654).
get filtered command list

## Test Plan and Hands on Testing

- upload a protocol that will enter ER.
- add some fixit commands.
- make sure you dont see the fixit commands in the run log. 

## Changelog

- added an arg `allCommands` to `get_run_commands`.
- add logic to `CommandState` and `CommandHistory` to filter based on the protocol intent. 

## Review requests

changes make sense? how ugly is this? 

## Risk assessment

Medium. need to make sure nothing that uses run commands changed the default behavior. 

[EXEC-654]: https://opentrons.atlassian.net/browse/EXEC-654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
